pkg/agent/openflow/network_policy.go

```diff
@@ -208,8 +209,15 @@ func (m *conjunctiveMatch) generateGlobalMapKey() string {
 		}
 	case net.IPNet:
 		valueStr = v.String()
+	case types.BitRange:
+		bitRange := m.matchValue.(types.BitRange)
+		if bitRange.Mask != nil {
+			valueStr = fmt.Sprintf("%d/%d", bitRange.Value, *bitRange.Mask)
+		} else {
+			valueStr = fmt.Sprintf("%d/nil", bitRange.Value)
```

Comment:
What will happen if I have a service with single port 17 and a service with port range 16-17. I guess the first service will generate 17/nil but the second will generate 16/0xffff and 17/0xffff. Will there be conflict between 17/nil and 17/0xffff? like if I remove port 17, the flow of 17/0xffff will be removed by accident?

Reply:
Good point. The situation you described will cause unexpected removal, but I think the reason is something else:

Based on my investigation, a rule with a 0xffff kind mask will be converted to a mask-less port rule in the OF table. For example, 16-18 we think it will be 0x0010/0xfffe and 0x0012/0xffff, but when checking the OF table it's 0x0010/0xfffe and 18. If one service is 16-18 and another service is 18, there will be only one 18 rule in the OF table. 

IMO, an overlap validation could make it easier and also save the time of realization of redundant rules.
Glad to hear others' thoughts.

Reply:
I think we mean the same. My assumption was different userspace cache key could cause same flow match field when the mask is optional, like IP/32 and IP. We have this kind of bug before and our solution is normalizing the userspace keys so that if their flow match field are same, their key in memory should be same too.

Reply:
See https://github.com/vmware-tanzu/antrea/blob/c58aef27ef1519f3aec4caf960a82ac95c67a5a4/pkg/agent/openflow/network_policy.go#L189-L202.
I think you should set full mask if it's single port so that 17/0xffff and 17/nil can get same key

Reply:
I got you. Fixed it. Thx.